### PR TITLE
AIRFLOW-51 && AIRFLOW-71 && AIRFLOW-516 docker_operator improvements

### PR DESCRIPTION
**Pull request by asnir**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:

[AIRFLOW-51] Improve the integration with swarm
[AIRFLOW-71] docker_operator - pulling from private repositories
[AIRFLOW-516] removing stopped container.

https://github.com/apache/incubator-airflow/pull/1806

### CODE REVIEW

File: airflow/operators/docker_operator.py [hunk 1]
```diff
@@ -124,21 +136,28 @@ def __init__(
...
         self.xcom_all = xcom_all
+        self.dockercfg_path = dockercfg_path
+        self.registry_username = registry_username
+        self.clear_volumes = clear_volumes
 
         self.cli = None
         self.container = None
 
+        if cpus is not None and isinstance(cpus, float):
+            logging.warning("cpus is deprecated property. Use cpu_shares instead.")
```
Comment (jlowin) [inline airflow/operators/docker_operator.py]: Please use a `DeprecationWarning` and update `UPDATING.md` appropriately

**Review by jlowin — CHANGES_REQUESTED** [review]

**Comment by bolkedebruin** [comment]
@asnir can you please rebase and fix the landscap/travs issues? Then we can merge (obviously with a proper commit message as per instructions plz)

**Comment by asnir** [comment]
@bolkedebruin, the CI process isn't stable. 
==============New===========================
p.s. please re-run the ci process, since it worked for my repository.

**Comment by bolkedebruin** [comment]
@asnir working on it see #1963

**Comment by bolkedebruin** [comment]
@asnir please try again, CI should be ok now.

**Comment by asnir** [comment]
I don't have permissions to re-run it.

**Comment by asnir** [comment]
@bolkedebruin , it even worse.

**Comment by bolkedebruin** [comment]
@asnir Try again plz. An update to a version jinja2 has a regression. I just fixed it by pinning the version.

**Comment by asnir** [comment]
@bolkedebruin

**Comment by bolkedebruin** [comment]
Nice now please sqUash your commits and provide a proper commit message per guidelines. That we can put it in the next release. Thanks for your patience.

**Comment by bolkedebruin** [comment]
hey @asnir I would really want to make this part of the 1.8.0 release. Can you squash your commits and create a commit message?

**Comment by asnir** [comment]
https://github.com/apache/incubator-airflow/pull/1982

**Pull request closed without merging**